### PR TITLE
Sobaner appgwt conditional rewrite

### PR DESCRIPTION
**Pull request by sobaner-zz**
## Description

Adding Support for RewriteRuleConditions under RewriteRule for Application Gateway Header CRUD Feature

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by vladimir-shcherbakov** [comment]
@sobaner Jenkins ci-build failed, the reason is here: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/11339/artifact/artifacts/ValidateHelpIssues.csv/*view*/

**Comment by MikhailTryakhov** [comment]
@sobaner there are errors and comments here. please fix them asap or you'll have to miss this release

**Comment by MikhailTryakhov** [comment]
deadline is EOD, please use ```19.10.0``` SDK version

**Comment by sobaner** [comment]
@vladimir-shcherbakov  - Thanks for pointing the error. Fixed it.
@MikhailTryakhov  - Fixed the issues and uploaded the new iteration. For this PR, my swagger is already checked-in during Jan 22 un january release branch and the design review for this PR also got accepted in January release branch only.

**Comment by vladimir-shcherbakov** [comment]
@sobaner you set the checkbox:
image

Could you add a link to the **cmdlet design review**?

**Comment by MikhailTryakhov** [comment]
@sobaner please provide design review and discuss it with Vlad ASAP. 
I'll send monthly branch to release at 1 pm

You can review our milestones, may be these dates can work for you

**Comment by MikhailTryakhov** [comment]
Closing due to design review wasn't confirmed/passed

**Comment by MikhailTryakhov** [comment]
please retarget to the next monthly branch when it's ready

**Comment by sobaner** [comment]
The Link of the Design Review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/234

File: src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1 [hunk 1]
```diff
@@ -769,6 +769,197 @@ function Test-ApplicationGatewayCRUDRewriteRuleSet
 	}
 }
 
+function Test-ApplicationGatewayCRUDRewriteRuleSetWithConditions
+{
+	param
+	(
+		$basedir = "./"
```
Comment (markcowl) [inline src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1]: You will need to get the actual base dir from the current AppDomain (e.g. ```AppDomain.CurrentDomain.BesDirectory```)

File: src/Network/Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallCRUD.json [hunk 2]
```diff
@@ -1252,10 +1252,10 @@
         ],
         "Date": [
           "Sat, 09 Mar 2019 22:38:08 GMT"
-        ],
+      
+        ],  ],
```
Comment (markcowl) [inline src/Network/Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallCRUD.json]: Looks like a serialization error with an incorrect character here

File: src/Network/Network/Models/PSApplicationGatewayRewriteRuleCondition.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+using Microsoft.Azure.Management.Network.Models;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Network.Models
+{
+    public partial class PSApplicationGatewayRewriteRuleCondition
+    {
+        public string Variable { get; set; }
```
Comment (markcowl) [inline src/Network/Network/Models/PSApplicationGatewayRewriteRuleCondition.cs]: This should have PS1XML tags, and a corresponding change in the format.ps1xml file for network - You want this type to appear in table format by default whenever it is output

File: src/Network/Network/help/New-AzApplicationGatewayRewriteRuleCondition.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+## DESCRIPTION
+**The AzApplicationGatewayRewriteRuleCondition** cmdlet creates a rewrite rule condition for an Azure application gateway.
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> $condition = New-AzApplicationGatewayRewriteRuleCondition -Variable "var_request_uri" -Pattern "http" -IgnoreCase
```
Comment (markcowl) [inline src/Network/Network/help/New-AzApplicationGatewayRewriteRuleCondition.md]: You should show the output when $condition is written to the screen as well.

**Comment by vladimir-shcherbakov** [comment]
@MikhailTryakhov 
Travis CI is failing: https://travis-ci.org/Azure/azure-powershell/jobs/509103930#L2207

File: src/Network/Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallCRUD.json (hunk 2 above)
Comment (markcowl) [reply src/Network/Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallCRUD.json]: This is most likely what is causing your test failure.

**Comment by MikhailTryakhov** [comment]
@sobaner  there are comments here related to changes in code. can you please answer/fix them, fix merging issue mentioned below (or revert my merge and create your own) so hope it'll unblock PR for merging

File: src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1 (hunk 1 above)
Comment (number213) [reply src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1]: @markcowl the test is always invoked with BaseDirectory, see `ApplicationGatewayTests.cs`
```csharp
TestRunner.RunTestScript(string.Format("Test-ApplicationGatewayCRUDRewriteRuleSetWithConditions -baseDir '{0}'", AppDomain.CurrentDomain.BaseDirectory));
```

File: src/Network/Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallCRUD.json (hunk 2 above)
Comment (number213) [reply src/Network/Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallCRUD.json]: Fixed

File: src/Network/Network/help/New-AzApplicationGatewayRewriteRuleCondition.md (hunk 4 above)
Comment (number213) [reply src/Network/Network/help/New-AzApplicationGatewayRewriteRuleCondition.md]: Done

File: src/Network/Network/Models/PSApplicationGatewayRewriteRuleCondition.cs (hunk 3 above)
Comment (number213) [reply src/Network/Network/Models/PSApplicationGatewayRewriteRuleCondition.cs]: Done

**Comment by number213** [comment]
@markcowl fixed CI and addressed your comments, please take a look

**Comment by MikhailTryakhov** [comment]
LGTM, thanks a lot @number213

**Pull request merged**